File: leetcode/go/590.n-ary-tree-postorder-traversal.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func postorder(root *Node) []int {
    post := []int{}
    dfs(root, &post)
    return post
}

func dfs(node *Node, post *[]int) {
    if node == nil {
        return
    }
    if len(node.Children) > 0 {
        for _, c := range node.Children {
            dfs(c, post)
        }
    }
    *post = append(*post, node.Val)
}

File: leetcode/python/590.n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        def dfs(root, arr):
            if root is None:
                return
            for child in root.children:
                dfs(child, arr)
            arr.append(root.val)
        ans = []
        dfs(root, ans)
        return ans

